compiler/src/main/kotlin/ast/reduce.kt

```diff
@@ -29,114 +32,112 @@ import org.antlr.v4.runtime.tree.TerminalNode
+            ctx = SourceContext(node),
             left = reduceExpr(node.left),
             right = reduceExpr(node.right)
         )
-        is CellmataParser.LessEqExprContext -> LessEqExpr(
-            ctx = node,
+        is CellmataParser.LessEqExprContext -> LessOrEqExpr(
+            ctx = SourceContext(node),
             left = reduceExpr(node.left),
             right = reduceExpr(node.right)
         )
         is CellmataParser.LessExprContext -> LessThanExpr(
-            ctx = node,
+            ctx = SourceContext(node),
             left = reduceExpr(node.left),
             right = reduceExpr(node.right)
         )
         is CellmataParser.AdditionExprContext -> AdditionExpr(
-            ctx = node,
+            ctx = SourceContext(node),
             left = reduceExpr(node.left),
             right = reduceExpr(node.right)
         )
         is CellmataParser.SubstractionExprContext -> SubtractionExpr(
-            ctx = node,
+            ctx = SourceContext(node),
             left = reduceExpr(node.left),
             right = reduceExpr(node.right)
         )
         is CellmataParser.MultiplictionExprContext -> MultiplicationExpr(
-            ctx = node,
+            ctx = SourceContext(node),
             left = reduceExpr(node.left),
             right = reduceExpr(node.right)
         )
         is CellmataParser.DivisionExprContext -> DivisionExpr(
-            ctx = node,
+            ctx = SourceContext(node),
             left = reduceExpr(node.left),
             right = reduceExpr(node.right)
         )
         is CellmataParser.ModuloExprContext -> ModuloExpr(
-            ctx = node,
+            ctx = SourceContext(node),
             left = reduceExpr(node.left),
             right = reduceExpr(node.right)
         )
-        is CellmataParser.NegativeExprContext -> NegativeExpr(
-            ctx = node,
+        is CellmataParser.NegationExprContext -> NegationExpr(
+            ctx = SourceContext(node),
             value = reduceExpr(node.value)
         )
-        is CellmataParser.InverseExprContext -> InverseExpr(
-            ctx = node,
+        is CellmataParser.NotExprContext -> NotExpr(
+            ctx = SourceContext(node),
             value = reduceExpr(node.value)
         )
         is CellmataParser.ArrayLookupExprContext -> ArrayLookupExpr(
-            ctx = node,
-            ident = reduceExpr(node.value),
+            ctx = SourceContext(node),
+            arr = reduceExpr(node.value),
             index = reduceExpr(node.index)
         )
-        is CellmataParser.ParenExprContext -> ParenExpr(
-            ctx = node,
-            expr = reduceExpr(node.expr())
-        ) // ToDo: Should we flatten this?
-        is CellmataParser.LiteralExprContext -> reduceExpr(node.value)
-        is CellmataParser.VarExprContext -> NamedExpr(
-            ctx = node,
-            ident = node.ident.text
+        is CellmataParser.ParenExprContext -> reduceExpr(node.expr())
```

Comment:
Are we sure it's safe to remove parenthesis?

Reply:
Yes. Parentheses just *change the precedence* and should only affect the parser, i.e. the parser builds a slightly different AST. But it shouldn't have any consequences.